src/util.ts

```diff
@@ -13,3 +14,36 @@ const Fork = fork(function*() {}).constructor;
 export function getCurrentExecution(): Execution {
   return Fork["currentlyExecuting"] as Execution;
 }
+
+/**
+ * Takes the common Node pattern of a callback with an error
+ * parameter, and wraps it to return a yieldable Operation
+ * instead.
+ */
+export function resumeOnCb(fn: (cb: (error: Error) => void) => void): Operation {
```

Comment:
This seems like such a common pattern that it should probably live inside Effection itself?

Reply:
I agree. The only reason that I've been hesitant to pull the trigger is that I'm not confident I fully understand all of the community conventions and patterns around callbacks and error handling.

I _think_ that most of the time, if there is an error, it is passed as the first argument, but sometimes this is not the case (as in `request.end()` in which the data is the first argument and errors get their own channel via `on("error)`). 

I do feel confident that with a little study we can come up with a few functions align with all of the use-cases and provide a guide of when to use each.

Reply:
Yes, let's let it live inside here for the time being and see how it feels :)